FWCore/Skeletons/test/runtests.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash -ex
+
+SCRAM_TEST_NAME=TestFWCoreSkeletons
+
+# Create a new CMSSW dev area
+OLD_CMSSW_BASE=$CMSSW_BASE
+rm -rf $SCRAM_TEST_NAME
+mkdir $SCRAM_TEST_NAME
+cd $SCRAM_TEST_NAME
+scram -a $SCRAM_ARCH project $CMSSW_VERSION
+cd $CMSSW_VERSION/src
+eval `scram run -sh`
+git cms-init
+
+# Copy FWCore/Skeletons in cause unit test is run during PR tests which contains changes for FWCore/Skeletons
+if [ -d $OLD_CMSSW_BASE/src/FWCore/Skeletons ] ; then
+    mkdir FWCore
+    cp -r $OLD_CMSSW_BASE/src/FWCore/Skeletons FWCore/Skeletons
+    scram build
+fi
+
+# Create TestSubsystem for testing the scripts
+mkdir TestSubsystem
+
+# Test output of scripts that can be easily compiled
+pushd TestSubsystem
+
+mkdqmedanalyzer TestDQMAnalyzerStream example_stream -author "Test Author"
+mkdqmedanalyzer TestDQMAnalyzerGlobal example_global -author "Test Author"
+mkdqmedanalyzer -debug TestDQMAnalyzerStrDbg example_stream -author "Test Author"
+mkdqmedanalyzer -debug TestDQMAnalyzerGlbDbg example_global -author "Test Author"
+
+mkedanlzr TestEDAnalyzer -author "Test Author"
+mkedanlzr TestEDAnalyzerHisto example_histo -author "Test Author"
+
+mkedfltr TestEDFilter -author "Test Author"
+
+mkedprod TestEDProducer -author "Test Author"
+mkedprod TestEDProducerParticle example_myparticle -author "Test Author"
+
+mkrecord TestRecord -author "Test Author"
+
+popd
+git add TestSubsystem
+git commit -a -m "Add skeleton modules that can be easily compiled"
+scram b code-checks-all
+scram b code-format-all
+git diff
+if [ $(git diff --name-only | grep TestSubsystem | wc -l) -gt 0 ] ; then
+    echo "code-checks or code-format caused differences, skeleton templates need to be updated!"
+    exit 1
+fi
```

Comment:
@makortel , may be we can remove these lines to run the tests bit faster ( we are running these any way later in the scipt).

Reply:
Do we trust that if the files before `code-{checks,format}-all` compile, they will compile also afterwards? Maybe the risk is negligible, and these are anyway examples and thus not critical.

Reply:
Pre-condition of code-checks/clang-tidy  is that code should compile. So yes we should make the first set of generated filed build error free.

Reply:
Removed now